src/resolve/resolved-manager.c

```diff
@@ -1124,18 +1124,15 @@ void manager_refresh_rrs(Manager *m) {
         }
 }
 
-int manager_next_hostname(Manager *m) {
+static int manager_next_random_name(const char *old, char **ret_new) {
         const char *p;
         uint64_t u, a;
-        char *h, *k;
-        int r;
+        char *n;
 
-        assert(m);
-
-        p = strchr(m->llmnr_hostname, 0);
+        p = strchr(old, 0);
         assert(p);
 
-        while (p > m->llmnr_hostname) {
+        while (p > old) {
                 if (!strchr("0123456789", p[-1]))
```

Comment:
actually, instead of "0123456789" we should probably use the DIGITS macro we already have